misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/InvocationArgProcessor.java

```diff
@@ -134,6 +158,136 @@
         return argValues;
     }
 
+    static Map<String, Value> generateNamedArgs(SuspendedContext context, String functionName, FunctionSignatureNode
+            functionSignature, List<Map.Entry<String, Evaluator>> argEvaluators) throws EvaluationException {
+
+        boolean namedArgsFound = false;
+        boolean restArgsFound = false;
+        List<ParameterNode> params = new ArrayList<>();
+        Map<String, ParameterNode> remainingParams = new HashMap<>();
+
+        if (!functionSignature.parameters().isEmpty()) {
+            for (ParameterNode paramNode : functionSignature.parameters()) {
+                params.add(paramNode);
+                remainingParams.put(getParameterName(paramNode), paramNode);
+            }
+        }
+
+        Map<String, Value> argValues = new HashMap<>();
+        for (int i = 0; i < argEvaluators.size(); i++) {
+            Map.Entry<String, Evaluator> arg = argEvaluators.get(i);
+            ArgType argType = getArgType(arg);
+            if (argType == ArgType.POSITIONAL) {
+                if (namedArgsFound) {
+                    throw new EvaluationException(String.format(EvaluationExceptionKind.CUSTOM_ERROR.getString(),
+                            "positional args are not allowed after named args."));
+                } else if (restArgsFound) {
+                    throw new EvaluationException(String.format(EvaluationExceptionKind.CUSTOM_ERROR.getString(),
+                            "positional args are not allowed after rest args."));
+                }
```

Comment:
Do we need a `else if` here? Instead shall we add
```
if (namedArgsFound || restArgsFound) {
    throw new EvaluationException()
}
```

Reply:
We cannot, as the Error messages are different.